pkg/util/http/client.go

```diff
@@ -31,86 +27,68 @@ func SanitizeURL(message string) string {
 	return apiKeyRegExp.ReplaceAllString(message, apiKeyReplacement)
 }
 
-// CreateHTTPTransport creates an *http.Transport for use in the agent
-func CreateHTTPTransport() *http.Transport {
-	tlsConfig := &tls.Config{
-		InsecureSkipVerify: config.Datadog.GetBool("skip_ssl_validation"),
+// Client wraps (http.Client).Do and resets the underlying connections at the
+// configured interval
+type Client struct {
+	httpClient    *http.Client
+	newHTTPClient func() *http.Client
+	resetInterval time.Duration
+
+	lastReset time.Time
+	mutex     sync.RWMutex
+}
+
+// NewClient returns an initialized Client resetting connections at the passed resetInterval.
+// The underlying http.Client used will be created using the passed http client factory.
+func NewClient(resetInterval time.Duration, newHTTPClient func() *http.Client) *Client {
```

Comment:
```suggestion
func NewClient(resetInterval time.Duration, clientFactory func() *http.Client) *Client {
```

Reply:
I do have one more q. Why a factory and not the client directly? Seems unusual.

Reply:
for full context: the `ResetClient` needs a factory and not a client directly because:

* the reset logic relies on re-creating a brand new client (to make sure there is no cached state such as cached connections remaining after a "reset", and that we can safely swap out the underlying client while concurrent operations are still taking place with the current client)
* clients don't offer a way to "clone" themselves
* there are slight differences between Logs's http client settings and APM's, and I don't want to try to normalize them in this PR